compiler/rustc_ast_lowering/src/asm.rs

```diff
@@ -374,11 +374,28 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                                         _ => None,
                                     };
 
+                                    let reg_str = |idx, default| -> &str {
+                                        let (op, _): &(InlineAsmOperand, Span) = &asm.operands[idx];
+                                        match op {
```

Comment:
```suggestion
                                        match asm.operands[idx].0 {
```

Reply:
Could you also add a comment saying that HIR asm doesn't preserve the original alias spelling, so we go to AST to get it?